### DISCUSSION TITLE
Add file transcription capability

### DISCUSSION THREAD

**Discussion by doxgt** [kdalanon/ChatGPT-AutoHotkey-Utility#9] [Ideas]
Great App!

Any chance you could consider adding the ability to upload a file for audio transcription through the [transcription API](https://platform.openai.com/docs/guides/speech-to-text/quickstart)?  

Thanks in advance for considering!

**Comment by kdalanon** [comment]
Hello,

That is an interesting idea, but for now I will not add that capability as I intended the program to focus more on text-based tasks.

**Reply by doxgt** [reply]
Thanks for replying.  Would you have any input on what the interface strings should look like for speech (https://platform.openai.com/docs/api-reference/audio)?  I tried to modify your function but couldn't get the speech API syntax right.  

I changed API_URL to "https://api.openai.com/v1/audio/transcriptions".

But I am stuck on the recorded file name syntax.

Anyway, if you have an thoughts on how I could modify things to make speech work, I'd love t hear it.  Thanks in advance!

Warm regards.

**Reply by kdalanon** [reply]
Hi doxgt,

I haven't played around with speech API, but upon glancing at the links you provided, it seems it's API is slightly different than chat, so you need to refactor the `HTTP_Request` part of my script, and not just change the `API_URL` to https://api.openai.com/v1/audio/transcriptions

**Reply by doxgt** [reply]
Thanks!

I kind of figured that but couldn't get the format/interface right.

I copied your main function to a:

`ProcessRequestTranscription(API_Model)`, since for transcription "API_Model" is the only parameter I would start out with (prompting can be added later from your examples).

Commented out section on ChatGPT_Prompt etc.

Made the following superficial changes:

```
;    Messages := '{ "role": "user", "content": "' ChatGPT_Prompt '" }'
    FileUpload := '{ "PATH/to/audiofile" }'
    JSON_Request := '{ "model": "' API_Model '", "file": [' FileUpload '] }'
```

Which didn't work.

I am wondering if I need to somehow send the raw bytes in the audiofile, encoded in Base64 or something, and then passed along to OpenAI in a JSON string?

That would seem complicated - to me that is.

Perhaps it would be easier to use AHK to call the Python script and using OpenAI's dedicated Python API interface?

**Reply by doxgt** [reply]
Turns out that using Curl in CMD through Run %ComSpec% in AHK is the easiest way to fulfill my purposes.
